pytestqt/plugin.py

```diff
@@ -308,7 +319,7 @@ def wait(self):
         if self.timeout is None and len(self._signals) == 0:
             raise ValueError("No signals or timeout specified.")
         if self.timeout is not None:
-            QtCore.QTimer.singleShot(self.timeout, self._loop.quit)
+            QtCore.QTimer.singleShot(self.timeout, self._quit_loop_by_timeout)
         self._loop.exec_()
```

Comment:
You are correct, if you raise during `_quit_loop_by_timeout`, that exception will be raised within Qt's event loop and will be captured. 

I think if you instead raise after `exec_()` it should work:

```python
self._loop.exec_()
if not self.signal_triggered and self.raising:
    raise QtBot.SignalTimeout(...)
```

Sorry I don't have time to test it myself right now, but I will try to review your PR more thoroughly tomorrow! :smile: